pkg/reconciler/reconcile.go

```diff
@@ -86,7 +86,11 @@ func (r *GenericDeploymentReconciler[T, L]) ReconcileAdapter(ctx context.Context
 	}
 	rcl.GetStatusManager().MarkSink(sinkURI)
 
-	desiredAdapter := ab.BuildAdapter(rcl, sinkURI)
+	desiredAdapter, err := ab.BuildAdapter(rcl, sinkURI)
+	if err != nil {
+		return controller.NewPermanentError(reconciler.NewEvent(corev1.EventTypeWarning,
+			ReasonInvalidSpec, "Could not generate desired state of adapter Deployment: %s", err))
+	}
```

Comment:
Error handled here as a fatal.
Rationale: if we can't generate the desired state of the adapter from the provided spec, the spec is most likely invalid.